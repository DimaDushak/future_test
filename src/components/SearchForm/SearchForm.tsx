import React from 'react';
import styles from './searchform.css';

interface ISearchFormProps {
    handleFilter: (value: string) => void;
}

export function SearchForm({ handleFilter }: ISearchFormProps) {
    const [ inputValue, setInputValue ] = React.useState('');

    const onSubmit = (e: React.FormEvent<HTMLFormElement>) => {
        e.preventDefault();
        handleFilter(inputValue.trim());
    }

    return (
        <form onSubmit={onSubmit}>
            <input
                placeholder="Поиск"
                value={inputValue}
                onChange={(e) => setInputValue(e.currentTarget.value)}
            />
            <button className={styles.button}>Найти</button>
        </form>
    );
}
