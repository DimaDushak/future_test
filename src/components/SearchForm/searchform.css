.button {
    margin-left: 10px;
    padding: 0 5px;
}
