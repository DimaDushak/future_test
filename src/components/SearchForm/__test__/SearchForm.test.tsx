import React from 'react';
import { shallow } from 'enzyme';
import { SearchForm } from '../SearchForm';

describe('SearchForm', () => {
    const wrapper = shallow(<SearchForm handleFilter={() => {}} />);
    const string = '125';

    test('should render (snapshot)', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('render after onChange', () => {
        wrapper.find('input').simulate('change', {
            currentTarget: {
                value: string
            }
        });

        expect(wrapper).toMatchSnapshot();
    });
});
