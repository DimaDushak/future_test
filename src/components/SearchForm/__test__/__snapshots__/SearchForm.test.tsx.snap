// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm render after onChange 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Поиск"
    value="125"
  />
  <button
    className="button"
  >
    Найти
  </button>
</form>
`;

exports[`SearchForm should render (snapshot) 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Поиск"
    value=""
  />
  <button
    className="button"
  >
    Найти
  </button>
</form>
`;
